cli/js/tests/unit_test_runner.ts

```diff
@@ -312,11 +295,10 @@ async function main(): Promise<void> {
 
   // Running tests matching current process permissions
   await registerUnitTests();
-  await Deno.runTests({
-    failFast: false,
-    exitOnFail: true,
-    only: filter
-  });
+  for await (const _ of Deno.runTests({
+    only: filter,
+    reportToConsole: true
+  }));
```

Comment:
So you can't do `await Deno.runTests()` anymore?

Reply:
> So you can't do `await Deno.runTests()` anymore?

Nope. I don't think `Deno.runTests()`'s ergonomics is super important at the moment... the way it's currently designed means you can't use it with `deno test` and it makes you run a test suite with `deno run`, it's hard to not see it as a niche technical thing.

Eventually I would propose making `Deno.runTests()` internal and replacing it with something hooks based that can affect `deno test`, e.g.
```ts
/** Execute the given function for each message received from the coming `deno test` run.
  * If it returns a string, this is used as the report output.
  */
Deno.setOnTestMessage(f: (msg: Deno.TestMessage) => string | null | Promise<string | null>);
```

Reply:
> > So you can't do `await Deno.runTests()` anymore?
> 
> Nope. I don't think `Deno.runTests()`'s ergonomics is super important at the moment... the way it's currently designed means you can't use it with `deno test` and it makes you run a test suite with `deno run`, it's hard to not see it as a niche technical thing.
> 
> Eventually I would propose making `Deno.runTests()` internal and replacing it with something hooks based that can affect `deno test`, e.g.
> 
> ```ts
> /** Execute the given function for each message received from the coming `deno test` run.
>   * If it returns a string, this is used as the report output. */
> Deno.setOnTestMessage(f: (msg: Deno.TestMessage) => string | null | Promise<string | null>);
> ```

I don't understand this argument. `Deno.runTests()` triggers running all tests registered using `Deno.test()`. @ry proposed to get rid of `reporter` argument and replace it with a single function; so why not do:
```
Deno.runTests({
   failFast: ...
   // replace "reporter" argument
   onMsg: (msg) => void
});
```
without changing signature of the function.

Reply:
Yeah that wasn't a counter-argument or anything, I got side-tracked :p

I got rid of it to demonstrate that the `TestReporter` typedef isn't necessary and `Deno.runTests()`'s associated API can be made smaller, but I can revert https://github.com/denoland/deno/pull/4451/commits/a876476ec01306265386a7bde773b0c8372403a1.

Reply:
@nayeemrmn I do believe that async generator for `Deno.runTests()` is useful (exactly the examples you provided), but it should also be promise-like. (See TODOs I left in `testing.ts` about this bit). Let's revert it for now and address in follow up